src/test/java/com/ministore/api/integration/CreateUserTest.java

```diff
@@ -51,4 +51,39 @@ void mustBeAbleToCreateAUser() throws Exception
             .andExpect(jsonPath("$.id").value(response.id))
             .andExpect(jsonPath("$.username").value(response.username));
     }
+
+    @Test
+    void mustReturnNotFoundWhenUserNotFound() throws Exception {
+        this.mockMvc.perform(get("/user/abc"))
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    void mustReturnBadRequestWhenCreateUserDoesntReceivePayload() throws Exception {
+        this.mockMvc.perform(
+            post("/user")
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void mustReturnBadRequestWhenUsernameIsInvalid() throws Exception {
+        this.mockMvc.perform(
+            post("/user")
```

Comment:
Ah, just another nitpick I should have seen earlier:

Rest URIs represent collections of resources on the server, so they should be written as plural. So the right URI here would be `/users` instead of `/user`.